General Quality of Life Edits

mayaraman19 opened the pull request:
I am deeply sorry for pushing this right before school starts. This PR comprises of a lot of small edits including:

#### Intro
- Made less wordy

#### First, Neural Networks
- Made less wordy (kinda)
- Added links to slides to take up awkward space

#### Shining a Flashlight on Filters
- The flashlight card was compressed from two cards to one
- "If you guessed a koala..." part was made less wordy (mostly) and added koala image on the side

#### Fully-connected Neural Networks
- removed orphan components, made less wordy but now that I'm looking it still seems wordy

#### How do we use filters to analyze images?
- made less wordy

#### Filter picker
- tried to make it fully fit in viewport for as long as possible

#### Filter sizes
- Turned it into a back/next button situation rather than clicking the buttons

### Issues/Leftovers:
#### Intro
- The intro-panel image still needs to be changed?

#### Shining a Flashlight on Filters
- `e.preventDefault()` still isn't working as intended it seems (as in preventing scrolling behavior on touch screens)!

#### How do we use filters to analyze images?
- Looks kinda terrible on wider screens

#### Filter picker
- sizing of filter matrix/lines/squares are kinda awkward

#### More filters
- Need an actual title

#### General
- We are currently considering a style overhaul

mayaraman19 commented:
i do not know why it includes 10000 commits

mayaraman19 commented:
I'm trying to get the flashlight component to work on mobile, but it still picks up scrolling behavior, making the flashlight really hard to use because the website scrolls at the same time. I added e.preventDefault(), but I'm not sure if I'm using it in the right place. Would appreciate any input <3

krashanoff commented:
Hey Maya! Thanks for addressing the feedback given. I can't give a cohesive review at the moment, but I *can* provide some answers to your questions:

> i do not know why it includes 10000 commits

Most of these are merges of your remote tracking branch into your local (i.e., updating your local copy of the repository). Don't worry too much about them, they can be compressed down into a single commit with a squash merge.

> it still picks up scrolling behavior, making the flashlight really hard to use because the website scrolls at the same time.

There are three primary events with mobile touches that you would need to account for in the flashlight: `touchstart`, `touchmove`, and `touchend`. Each of these you would need to call `e.preventDefault()` at the start of to ensure that the scrolling does not occur. Alternatively, you could use CSS [as suggested in this thread](https://stackoverflow.com/a/43275544) to prevent the behavior on the component.

mayaraman19 commented:
I am deeply sorry for pushing this right before school starts. This PR comprises of a lot of small edits including:

#### Intro
- Made less wordy

#### First, Neural Networks
- Made less wordy (kinda)
- Added links to slides to take up awkward space

#### Shining a Flashlight on Filters
- The flashlight card was compressed from two cards to one
- "If you guessed a koala..." part was made less wordy (mostly) and added koala image on the side

#### Fully-connected Neural Networks
- removed orphan components, made less wordy but now that I'm looking it still seems wordy

#### How do we use filters to analyze images?
- made less wordy

#### Filter picker
- tried to make it fully fit in viewport for as long as possible

#### Filter sizes
- Turned it into a back/next button situation rather than clicking the buttons

### Issues/Leftovers:
#### Intro
- The intro-panel image still needs to be changed?

#### Shining a Flashlight on Filters
- `e.preventDefault()` still isn't working as intended it seems (as in preventing scrolling behavior on touch screens)!

#### How do we use filters to analyze images?
- Looks kinda terrible on wider screens

#### Filter picker
- sizing of filter matrix/lines/squares are kinda awkward

#### More filters
- Need an actual title

#### General
- We are currently considering a style overhaul

malsf21 reviewed (commented):
Did a side-by-side comparison and I think in general, this is a solid improvement in quite a few different areas, nothing to complain about! Going to go over the code in-depth tomorrow; thank you for all your work on this so far Maya!

malsf21 commented on src/components/FilterSizes.js:
```diff
@@ -8,58 +8,100 @@ class FilterSizes extends React.Component {
         <section className = "blue section">
             <div className="card">
-                <h2>click to see what's up</h2>
+                <h2>Yooooo</h2>
```
Is this supposed to be here 👀

malsf21 commented on src/components/FilterSizes.js:
```diff
@@ -8,58 +8,100 @@ class FilterSizes extends React.Component {
+            ["78%", "81%"]
         ];
         
         const buttons = params.map((val, index) => {
```
These aren't buttons anymore, right? Do you mind changing the variable name for clarity then?

malsf21 commented on src/components/FilterSizes.js:
```diff
@@ -8,58 +8,100 @@ class FilterSizes extends React.Component {
     constructor(props) {
         super(props);
         this.state = {
+            paragraphs: [ //text, style.top, style.left
```
Is there a reason that this isn't a constant? Do the paragraphs change?

mayaraman19 replied:
they don't change; i've moved it out of state

malsf21 commented on src/components/Filters.js:
```diff
@@ -99,47 +99,48 @@ class Filters extends React.Component {
+                <p>Select the different filters in this example. What patterns do you see in the filter values?</p>
+            </div>
+            <div className="filters-holder">
                 <img src={amsterdamRegular} alt="black and white amsterdam" width="500px"></img>
```
Ooh, I didn't catch this earlier, but I would avoid hard-coding the `width` as an HTML attribute - it's confusing when combined with CSS classes!

malsf21 commented on src/components/Filters.js:
```diff
@@ -99,47 +99,48 @@ class Filters extends React.Component {
+                        </label>
+                    </form>
                 </div>
+                <img id="filter-applied" src={this.state.src} alt="amsterdam with filter applied" width="500px"></img>
```
Same comment on hard-coded width

malsf21 commented on src/components/Filters.js:
```diff
@@ -99,47 +99,48 @@ class Filters extends React.Component {
                     </div>
-                    <img id="filter-applied" src={this.state.src} alt="amsterdam with filter applied" width="500px"></img>
+                    <form>
+                        <label>
+                            Choose a filter:
+                            <select value={this.state.value} onChange={this.handleChange}>
+                                <option value="identity">Identity</option>
+                                <option value="blur">Blur</option>
+                                <option value="bottom sobel">Bottom sobel</option>
+                                <option value="emboss">Emboss</option>
+                                <option value="left sobel">Left sobel</option>
+                                <option value="outline">Outline</option>
+                                <option value="right sobel">Right sobel</option>
+                                <option value="sharpen">Sharpen</option>
+                                <option value="top sobel">Top sobel</option>
+                            </select>
+                        </label>
```
Also disappointed that I didn't catch this myself earlier, but AFAIK the `<label>` and `<select>` are neighbour tags. See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select

malsf21 commented on src/components/Flashlight.js:
```diff
@@ -7,7 +7,8 @@ import './style/Flashlight.css';
-    showFlashlight = () => {
+    showFlashlight = e => {
+        e.preventDefault();
         const spotlight = document.getElementById("spotlight");
```
Avoid using direct DOM manipulation when possible, though I know the flashlight component is a bit tricky. Not merge-blocking for now, but something we should refactor!

malsf21 commented on src/components/Flashlight.js:
```diff
@@ -7,7 +7,8 @@ import './style/Flashlight.css';
 
-    showFlashlight = () => {
+    showFlashlight = e => {
+        e.preventDefault();
```
I can figure this out with you later (the mobile compat), for now let's table it (but maybe add a `// TODO` comment)

malsf21 commented on src/components/Flashlight.js:
```diff
@@ -16,19 +17,38 @@ class Flashlight extends React.Component {
-        const spotlight = document.getElementById("spotlight");
+    hoverMoveFlashlight = e => {
+        e.preventDefault();
         const container = document.getElementById("container");
```
Same DOM manipulation comment, `// TODO`

malsf21 commented on src/components/Flashlight.js:
```diff
@@ -16,19 +17,38 @@ class Flashlight extends React.Component {
+
+    touchMoveFlashlight = e => {
+        e.preventDefault();
+        const container = document.getElementById("container");
+        let rect = container.getBoundingClientRect();
```
Same DOM manipulation comment, `// TODO`

malsf21 commented on src/components/Flashlight.js:
```diff
@@ -16,19 +17,38 @@ class Flashlight extends React.Component {
+        let rect = container.getBoundingClientRect();
+
+        //get coordinates of mouse
+        let x = Math.round(e.targetTouches[0].clientX - rect.x - 20);
+        let y = Math.round(e.targetTouches[0].clientY - rect.y - 20);
```
Do you have to round it here? IRRC floats are supported with pixel values

malsf21 commented on src/components/Flashlight.js:
```diff
@@ -16,19 +17,38 @@ class Flashlight extends React.Component {
+    }
+
+    moveFlashlight = (x, y) => {
+        const spotlight = document.getElementById("spotlight");
```
Same DOM manipulation comment, `// TODO`

malsf21 commented on src/components/Flashlight.js:
```diff
@@ -16,19 +17,38 @@ class Flashlight extends React.Component {
-    hideFlashlight = () => {
+    hideFlashlight = e => {
+        e.preventDefault();
         const spotlight = document.getElementById("spotlight");
```
Same DOM manipulation comment, `// TODO`

malsf21 reviewed (approved):
Left some minor feedback, but I don't think anything's merge-blocking per se - knock out however much you feel like doing now, but you're always free to punt things to later. If so, just do me a favour and add a `// TODO` comment so we don't forget!

Another note: there is quite a few classes that are heavily duplicated (e.g. `color` or `background-color` for the AI blue, some flex utilities). Not merge-blocking either, but when you do end up doing the redesign - it may be helpful to consolidate some of those classes into reusable components, which should be easier to maintain.

Still, major respect for hitting on so much feedback @mayaraman19! Really appreciate it :) 

(oh, and please squash merge - there's quite a few commits!)
